File: Quest/Assets/Src/Game/QuestStage.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class QuestStage : Targetable
{
    private QuestInfo quest;

    List<AdventureCard> cards = new List<AdventureCard>();

    public QuestStage(QuestInfo info)
    {
        this.quest = info;
    }


    public void addCard(AdventureCard card)
    {
        cards.Add(card);
    }

    public int getBattlePoints(GameState state)
    {
        int total = 0;
        foreach (AdventureCard card in cards)
            total += card.getBattlePoints(state);
        return total;
    }

    public AdventureCard[] getCards() { return cards.ToArray(); }
}

File: eclipse/Quest/src/model/game/QuestStage.java
package model.game;

import java.util.ArrayList;

import model.cards.AdventureCard;

public class QuestStage {

	private QuestInfo quest;
	private int stage;
	
	ArrayList<AdventureCard> cards = new ArrayList<>();
	
	public QuestStage(QuestInfo info, int stage) {
		this.quest = info;
		this.stage = stage;
	}
	
	
	public void addCard(AdventureCard card) {
		cards.add(card);
	}
	
	public int getBattlePoints(GameState state) {
		int total = 0;
		for (AdventureCard card : cards)
			total += card.getBattlePoints(state);
		return total;
	}
	
	public int getStageNum() { return stage; }
	
	public AdventureCard[] getCards() { return cards.toArray(new AdventureCard[cards.size()]); }
}
